### DISCUSSION TITLE
How can I change the DPI setting?

### DISCUSSION THREAD

**Discussion by NessaRNG** [owl4ce/dotfiles#172] [Q&A] [answered]
hello, i use your dotfiles is nice to see but the dpi is too small how i can change it ? i tried with xrandr and anything still changes nothing
![2022-04-11-175715_1920x1080_scrot](https://user-images.githubusercontent.com/103401388/162662848-f0be0401-e65b-44a5-b935-a77f0258c9f1.png)
g

**Comment by owl4ce** [comment, marked as answer]
**Hi @NessaRNG**

> Which version of dotfiles are you using? I see it looks like the previous dotfiles (v3).

Well, actually HiDPI is a bit tricky, I'm currently working on a good way to set it up, but for **rxvt-unicode** (URxvt) itself, you can change it by changing the configuration as follows.

https://github.com/owl4ce/dotfiles/commit/97a7f4fbb900bc0e558aaae9c7b5f895b4fe6c65
https://github.com/owl4ce/dotfiles/blob/97a7f4fbb900bc0e558aaae9c7b5f895b4fe6c65/.Xresources#L5-L8

Then for **rofi**. This is universally applied.

https://github.com/owl4ce/dotfiles/commit/5783aafa0538577fb1851ce9932d62a2da1baa85
https://github.com/owl4ce/dotfiles/blob/97a7f4fbb900bc0e558aaae9c7b5f895b4fe6c65/.config/rofi/config.rasi#L18

Unfortunately for the **tint2** panel itself it's possible to be relative, but the problem is that the padding is also relative so the shape is irregular at different DPIs. If you wish, change this line to scale relatively to specific DPI as a reference, just set it to 96. This also applies to every configuration, so change all the tint2 configurations.

https://github.com/owl4ce/dotfiles/blob/97a7f4fbb900bc0e558aaae9c7b5f895b4fe6c65/.config/tint2/mechanical-vertical.artistic.tint2rc#L154

Now the problem is application layers such as **GTK**, actually quite easy. Well, GTK uses Xft (like the URxvt font backend we use), so it can be configured via X resource database (xrdb).

The following command displays the current GTK DPI information. Mine is 96 by default.

```sh
💲 gtk-query-settings dpi
```

```sh
                        gtk-xft-dpi: 98304
```

Then to set it, add the following line to `~/.Xresources` to make it permanent.

```cfg
Xft.dpi: 120
```

Reload if you want to apply it without logging out.

```sh
💲 xrdb ~/.Xresources
```

If you prefer **xrandr**, it can be set to [start automatically](https://wiki.archlinux.org/title/Autostarting#On_Xorg_startup) by adding the xrandr command to `~/.xinitrc` if using **xinit**, and `~/.xprofile` for the display manager. For more information [here](https://wiki.archlinux.org/title/xorg#Setting_DPI_manually) and [here](https://wiki.archlinux.org/title/HiDPI).

**Reply by owl4ce** [reply]
For desktop notifications, **dunst** already respect DPI setting by default.

**Reply by owl4ce** [reply]
Scaling the **tint2** panel relatively also breaks openbox margin rules.

**Reply by NessaRNG** [reply]
okay it's worked thanks
